### PR TITLE
Add unit test for OtherParamsSelection has more than one bit set.

### PR DESCRIPTION
**Pull request by Zhiqiang520**
Fix: #1542

Signed-off-by: Zhao, Zhiqiang <zhiqiang.zhao@intel.com>

### CODE REVIEW

File: library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c [hunk 1]
```diff
@@ -193,6 +193,12 @@ static libspdm_return_t libspdm_try_negotiate_algorithms(libspdm_context_t *spdm
         status = LIBSPDM_STATUS_INVALID_MSG_FIELD;
         goto receive_done;
     }
+    if (spdm_response->header.spdm_version >= SPDM_MESSAGE_VERSION_12) {
+        if (!libspdm_onehot0(spdm_response->other_params_support)) {
```
Comment (steven-bellock) [inline library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c:197]: The name of the response field should be `other_params_selection` but I'll file a separate issue for that.

**Review by steven-bellock — APPROVED** [review]

File: library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c (hunk 1 above)
Comment (jyao1) [inline library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c:197]: it should be `if (!libspdm_onehot0(spdm_response->other_params_support & 0xF)) {`

The upper bit are reserved and should be ignored.

Comment (Zhiqiang520) [reply library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c]: The line 486~488 check the opaque data format must be **OpaqueDataFmt1**, but this PR is to check **no more than one bit** shall be set for the opaque data format.
https://github.com/DMTF/libspdm/blob/befd3a8c41427feb28719db4324ec2f82ed62790/library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c#L485-L488

Comment (jyao1) [reply library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c]: If it is only Fmt1, that already meets no more than one bit requirement.

I dont think more check is needed.

Comment (Zhiqiang520) [reply library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c]: ok, no more check is needed, and just add unit test for OtherParamsSelection has more than one bit set.

**Pull request merged**